File: Set row height and column width for all cells in an Excel sheet using automation/[C#][VB]-Set row height and column width for all cells in an Excel sheet using automation/C# and VB/ExcelLibrary_cs/Operations.cs
using System;
using Excel = Microsoft.Office.Interop.Excel;

namespace ExcelLibrary_cs
{
    public class Operations
    {
        /// <summary>
        /// File name to work with
        /// </summary>
        /// <value></value>
        /// <returns></returns>
        /// <remarks></remarks>
        public string FileName { get; set; }

        /// <summary>
        /// Sheet name to work with
        /// </summary>
        /// <value></value>
        /// <returns></returns>
        /// <remarks></remarks>
        public string SheetName { get; set; }

        /// <summary>
        /// Value to set all row's heights
        /// </summary>
        /// <value></value>
        /// <returns></returns>
        /// <remarks></remarks>
        public int RowHeight { get; set; }

        /// <summary>
        /// Value to set all column widths
        /// </summary>
        /// <value></value>
        /// <returns></returns>
        /// <remarks></remarks>
        public int ColumnHeight { get; set; }

        public Operations()
        {
        }

        public Operations(string FileName, string SheetName, int RowHeight, int ColumnHeight)
        {
            this.FileName = FileName;
            this.SheetName = SheetName;
            this.RowHeight = RowHeight;
            this.ColumnHeight = ColumnHeight;
        }

        public OperationReasons SetRowHeightColumnWidth()
        {
            if (System.IO.File.Exists(FileName))
            {
                bool SheetNotLocated = false;

                bool Proceed = false;

                Excel.Application xlApp = null;
                Excel.Workbooks xlWorkBooks = null;
                Excel.Workbook xlWorkBook = null;
                Excel.Worksheet xlWorkSheet = null;
                Excel.Sheets xlWorkSheets = null;
                Excel.Range xlCells = null;

                xlApp = new Excel.Application();
                xlApp.DisplayAlerts = false;

                xlWorkBooks = xlApp.Workbooks;
                xlWorkBook = xlWorkBooks.Open(FileName);

                xlApp.Visible = false;

                xlWorkSheets = xlWorkBook.Sheets;

                for (int x = 1; x <= xlWorkSheets.Count; x++)
                {
                    xlWorkSheet = (Excel.Worksheet)xlWorkSheets[x];

                    if (xlWorkSheet.Name == SheetName)
                    {
                        Proceed = true;
                        break;
                    }

                    System.Runtime.InteropServices.Marshal.FinalReleaseComObject(xlWorkSheet);
                    xlWorkSheet = null;
                }

                if (Proceed)
                {
                    xlCells = xlWorkSheet.Cells;
                    Excel.Range EntireRow = xlCells.EntireRow;
                    EntireRow.RowHeight = RowHeight;
                    EntireRow.ColumnWidth = ColumnHeight;

                    ReleaseComObject(xlCells);
                    ReleaseComObject(EntireRow);
                }
                else
                {
                    SheetNotLocated = true;
                }

                xlWorkSheet.SaveAs(FileName);

                xlWorkBook.Close();
                xlApp.UserControl = true;
                xlApp.Quit();

                ReleaseComObject(xlWorkSheets);
                ReleaseComObject(xlWorkSheet);
                ReleaseComObject(xlWorkBook);
                ReleaseComObject(xlWorkBooks);
                ReleaseComObject(xlApp);
                if (SheetNotLocated)
                {
                    return OperationReasons.SheetNotFound;
                }
                else
                {
                    return OperationReasons.Success;
                }
            }
            else
            {
                return OperationReasons.FileNameFound;
            }
        }

        /// <summary>
        /// Clean up objects
        /// </summary>
        /// <param name="obj"></param>
        /// <remarks></remarks>
        private void ReleaseComObject(object obj)
        {
            try
            {
                if (obj != null)
                {
                    System.Runtime.InteropServices.Marshal.ReleaseComObject(obj);
                    obj = null;
                }
            }
            catch (Exception ex)
            {
                obj = null;
            }
        }
    }

    public enum OperationReasons
    {
        /// <summary>
        /// Operation was sucessful
        /// </summary>
        /// <remarks></remarks>
        Success,

        /// <summary>
        /// File does not exists
        /// </summary>
        /// <remarks></remarks>
        FileNameFound,

        /// <summary>
        /// Sheet does not exists
        /// </summary>
        /// <remarks></remarks>
        SheetNotFound
    }
}

File: Set row height and column width for all cells in an Excel sheet using automation/[C#][VB]-Set row height and column width for all cells in an Excel sheet using automation/C# and VB/ExcelLibrary_vb/Operations.vb
Option Strict On
Imports Excel = Microsoft.Office.Interop.Excel
Imports Microsoft.Office
Imports System.Runtime.InteropServices
''' <summary>
''' Provides a method to set all row height and column width for an entire sheet.
''' </summary>
''' <remarks></remarks>
Public Class Operations
    ''' <summary>
    ''' File name to work with
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property FileName As String
    ''' <summary>
    ''' Sheet name to work with
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property SheetName As String
    ''' <summary>
    ''' Value to set all row's heights
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property RowHeight As Integer
    ''' <summary>
    ''' Value to set all column widths
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property ColumnHeight As Integer

    Public Sub New()

    End Sub
    Public Sub New(ByVal FileName As String, ByVal SheetName As String, ByVal RowHeight As Integer, ByVal ColumnHeight As Integer)
        Me.FileName = FileName
        Me.SheetName = SheetName
        Me.RowHeight = RowHeight
        Me.ColumnHeight = ColumnHeight
    End Sub
    Public Function SetRowHeightColumnWidth() As OperationReasons
        If IO.File.Exists(FileName) Then
            Dim SheetNotLocated As Boolean = False

            Dim Proceed As Boolean = False

            Dim xlApp As Excel.Application = Nothing
            Dim xlWorkBooks As Excel.Workbooks = Nothing
            Dim xlWorkBook As Excel.Workbook = Nothing
            Dim xlWorkSheet As Excel.Worksheet = Nothing
            Dim xlWorkSheets As Excel.Sheets = Nothing
            Dim xlCells As Excel.Range = Nothing

            xlApp = New Excel.Application
            xlApp.DisplayAlerts = False


            xlWorkBooks = xlApp.Workbooks
            xlWorkBook = xlWorkBooks.Open(FileName)

            xlApp.Visible = False

            xlWorkSheets = xlWorkBook.Sheets

            For x As Integer = 1 To xlWorkSheets.Count
                xlWorkSheet = CType(xlWorkSheets(x), Excel.Worksheet)

                If xlWorkSheet.Name = SheetName Then
                    Proceed = True
                    Exit For
                End If

                Runtime.InteropServices.Marshal.FinalReleaseComObject(xlWorkSheet)
                xlWorkSheet = Nothing

            Next

            If Proceed Then
                xlCells = xlWorkSheet.Cells
                Dim EntireRow As Excel.Range = xlCells.EntireRow
                EntireRow.RowHeight = RowHeight
                EntireRow.ColumnWidth = ColumnHeight

                ReleaseComObject(xlCells)
                ReleaseComObject(EntireRow)
            Else
                SheetNotLocated = True
            End If

            xlWorkSheet.SaveAs(FileName)

            xlWorkBook.Close()
            xlApp.UserControl = True
            xlApp.Quit()

            ReleaseComObject(xlWorkSheets)
            ReleaseComObject(xlWorkSheet)
            ReleaseComObject(xlWorkBook)
            ReleaseComObject(xlWorkBooks)
            ReleaseComObject(xlApp)
            If SheetNotLocated Then
                Return OperationReasons.SheetNotFound
            Else
                Return OperationReasons.Success
            End If
        Else
            Return OperationReasons.FileNameFound
        End If

    End Function
    ''' <summary>
    ''' Clean up objects
    ''' </summary>
    ''' <param name="obj"></param>
    ''' <remarks></remarks>
    Private Sub ReleaseComObject(ByVal obj As Object)
        Try
            If obj IsNot Nothing Then
                System.Runtime.InteropServices.Marshal.ReleaseComObject(obj)
                obj = Nothing
            End If
        Catch ex As Exception
            obj = Nothing
        End Try
    End Sub
End Class
Public Enum OperationReasons
    ''' <summary>
    ''' Operation was sucessful
    ''' </summary>
    ''' <remarks></remarks>
    Success
    ''' <summary>
    ''' File does not exists
    ''' </summary>
    ''' <remarks></remarks>
    FileNameFound
    ''' <summary>
    ''' Sheet does not exists
    ''' </summary>
    ''' <remarks></remarks>
    SheetNotFound
End Enum
